### PR TITLE
[FAB-17953] IT: add tests to verify channel config can be retrieved on…

### PR DESCRIPTION
**Pull request by wenjianqiao**
Signed-off-by: Wenjian Qiao <wenjianq@gmail.com>

#### Type of change
- New Feature
- Test update

#### Description
When a peer is bootstrapped from a snapshot, it does not have the config block(s)
before the snapshot. Discovery service and cscc have been updated to retrieve
channel confing from current state instead of the config block.
This PR adds e2e tests to verify it.

#### Related issues
https://jira.hyperledger.org/browse/FAB-17953

### CODE REVIEW

**Comment by wenjianqiao** [comment]
CI failed in /home/vsts/work/1/fabric/integration/raft/channel_participation_test.go:987
Probably related to https://github.com/hyperledger/fabric/pull/2065

**Comment by wlahti** [comment]
> CI failed in /home/vsts/work/1/fabric/integration/raft/channel_participation_test.go:987
> Probably related to #2065

I think I fixed the flake. Try rebasing on master when you get a chance. And FYI there's a snapshot flake now too... [FAB-18307](https://jira.hyperledger.org/browse/FAB-18307)

File: integration/discovery/discovery_test.go [hunk 1]
```diff
@@ -634,6 +654,97 @@ var _ = Describe("DiscoveryService", func() {
...
+			By("installing and instantiating chaincode")
+			chaincode := nwo.Chaincode{
+				Name:    "mycc",
+				Version: "1.0",
+				Path:    "github.com/hyperledger/fabric/integration/chaincode/simple/cmd",
+				Ctor:    `{"Args":["init","a","100","b","200"]}`,
+				Policy:  `OR (AND ('Org1MSP.member','Org2MSP.member'), AND ('Org1MSP.member','Org3MSP.member'), AND ('Org2MSP.member','Org3MSP.member'))`,
+			}
+			nwo.DeployChaincodeLegacy(network, "testchannel", orderer, chaincode, org1Peer0)
+
```
Comment (manish-sethi) [inline integration/discovery/discovery_test.go]: This shouldn't be required for this test... no?

File: integration/discovery/discovery_test.go [hunk 2]
```diff
@@ -634,6 +654,97 @@ var _ = Describe("DiscoveryService", func() {
 		Eventually(sess, network.EventuallyTimeout).Should(gexec.Exit(1))
 		Expect(sess.Err).To(gbytes.Say(`access denied`))
 	})
+
+	When("a peer is bootstrapped from a snapshot", func() {
```
Comment (manish-sethi) [inline integration/discovery/discovery_test.go]: Is it not possible to have this test at the end of the existing `It`, as you try to reduce the setup overheads at other places?

Comment (wenjianqiao) [reply integration/discovery/discovery_test.go]: Moved the snapshot test to an existing `It`.

File: integration/discovery/discovery_test.go (hunk 1 above)
Comment (wenjianqiao) [reply integration/discovery/discovery_test.go]: Not required

Comment (wenjianqiao) [reply integration/discovery/discovery_test.go]: Removed

**Review by manish-sethi — APPROVED** [review]

**Pull request merged**